### ISSUE TITLE
torch和mmcv的神奇搭配

### ISSUE THREAD

**Issue by WuBai2005** [LMMMEng/OverLoCK#55]
您好，我之前在RTX 4090上跑的时候是torch2.3.1+cu121（如下）（这是可以跑的），但是现在我打包到RTX A6000上在val阶段会报错，这是为什么(overlock) pxh@tom-Super-Server:~$ python -c "import mmcv; import mmdet; import torch; print(torch.__version__, mmcv.__version__, mmdet.__version__)"
/home/pxh/miniconda3/envs/overlock/lib/python3.10/site-packages/mmcv/__init__.py:20: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  warnings.warn(
2.3.1+cu121 1.7.2 2.28.2
(overlock) pxh@ac-Super-Server:~$ conda list
# packages in environment at /home/pxh/miniconda3/envs/overlock:
#
# Name                      Version               Build               Channel
_libgcc_mutex               0.1                   conda_forge         https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
_openmp_mutex               4.5                   2_gnu               https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
addict                      2.4.0                 pypi_0              pypi
aliyun-python-sdk-core      2.16.0                pypi_0              pypi
aliyun-python-sdk-kms       2.16.5                pypi_0              pypi
asttokens                   3.0.1                 pyhd8ed1ab_0        conda-forge
bzip2                       1.0.8                 hda65f42_8          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
ca-certificates             2026.2.25             hbd8a1cb_0          conda-forge
certifi                     2025.10.5             pypi_0              pypi
cffi                        2.0.0                 pypi_0              pypi
charset-normalizer          3.4.4                 pypi_0              pypi
click                       8.3.0                 pypi_0              pypi
colorama                    0.4.6                 pypi_0              pypi
comm                        0.2.3                 pyhe01879c_0        conda-forge
conda-pack                  0.9.1                 pyhcf101f3_0        conda-forge
contourpy                   1.3.2                 pypi_0              pypi
crcmod                      1.7                   pypi_0              pypi
cryptography                46.0.3                pypi_0              pypi
cycler                      0.12.1                pypi_0              pypi
debugpy                     1.8.18                py310h25320af_0     conda-forge
decorator                   5.2.1                 pyhd8ed1ab_0        conda-forge
einops                      0.8.1                 pyhd8ed1ab_0        https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
exceptiongroup              1.3.1                 pyhd8ed1ab_0        conda-forge
executing                   2.2.1                 pyhd8ed1ab_0        conda-forge
filelock                    3.14.0                pypi_0              pypi
fonttools                   4.60.1                pypi_0              pypi
fsspec                      2025.9.0              pypi_0              pypi
hf-xet                      1.1.10                pypi_0              pypi
huggingface-hub             0.35.3                pypi_0              pypi
idna                        3.11                  pypi_0              pypi
ipdb                        0.13.13               pypi_0              pypi
ipykernel                   7.1.0                 pyha191276_0        conda-forge
ipython                     8.37.0                pyh8f84b5b_0        conda-forge
jedi                        0.19.2                pyhd8ed1ab_1        conda-forge
jinja2                      3.1.6                 pypi_0              pypi
jmespath                    0.10.0                pypi_0              pypi
jupyter_client              8.8.0                 pyhcf101f3_0        conda-forge
jupyter_core                5.9.1                 pyhc90fa1f_0        conda-forge
keyutils                    1.6.3                 hb9d3cd8_0          conda-forge
kiwisolver                  1.4.9                 pypi_0              pypi
krb5                        1.21.3                h659f571_0          conda-forge
ld_impl_linux-64            2.44                  ha97dd6f_2          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libedit                     3.1.20250104          pl5321h7949ede_0    conda-forge
libexpat                    2.7.1                 hecca717_0          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libffi                      3.5.2                 h9ec8514_0          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgcc                      15.2.0                h767d61c_7          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgcc-ng                   15.2.0                h69a702a_7          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgomp                     15.2.0                h767d61c_7          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
liblzma                     5.8.1                 hb9d3cd8_2          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libnsl                      2.0.1                 hb9d3cd8_1          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libsodium                   1.0.20                h4ab18f5_0          conda-forge
libsqlite                   3.50.4                h0c1763c_0          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libstdcxx                   15.2.0                h8f9b012_7          conda-forge
libstdcxx-ng                15.2.0                h4852527_7          conda-forge
libuuid                     2.41.2                he9a06e4_0          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libxcrypt                   4.4.36                hd590300_1          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libzlib                     1.3.1                 hb9d3cd8_2          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
markdown                    3.9                   pypi_0              pypi
markdown-it-py              4.0.0                 pypi_0              pypi
markupsafe                  3.0.3                 pypi_0              pypi
matplotlib                  3.10.7                pypi_0              pypi
matplotlib-inline           0.2.1                 pyhd8ed1ab_0        conda-forge
mdurl                       0.1.2                 pypi_0              pypi
mmcv-full                   1.7.2                 pypi_0              pypi
mmdet                       2.28.2                pypi_0              pypi
mmengine                    0.2.0                 pypi_0              pypi
model-index                 0.1.11                pypi_0              pypi
mpmath                      1.3.0                 pypi_0              pypi
natten                      0.17.1+torch230cu121  pypi_0              pypi
ncurses                     6.5                   h2d0b736_3          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
nest-asyncio                1.6.0                 pyhd8ed1ab_1        conda-forge
networkx                    3.4.2                 pypi_0              pypi
ninja                       1.13.0                pypi_0              pypi
numpy                       2.2.6                 pypi_0              pypi
nvidia-cublas-cu12          12.1.3.1              pypi_0              pypi
nvidia-cuda-cupti-cu12      12.1.105              pypi_0              pypi
nvidia-cuda-nvrtc-cu12      12.1.105              pypi_0              pypi
nvidia-cuda-runtime-cu12    12.1.105              pypi_0              pypi
nvidia-cudnn-cu12           8.9.2.26              pypi_0              pypi
nvidia-cufft-cu12           11.0.2.54             pypi_0              pypi
nvidia-cufile-cu12          1.13.1.3              pypi_0              pypi
nvidia-curand-cu12          10.3.2.106            pypi_0              pypi
nvidia-cusolver-cu12        11.4.5.107            pypi_0              pypi
nvidia-cusparse-cu12        12.1.0.106            pypi_0              pypi
nvidia-cusparselt-cu12      0.7.1                 pypi_0              pypi
nvidia-nccl-cu12            2.20.5                pypi_0              pypi
nvidia-nvjitlink-cu12       12.8.93               pypi_0              pypi
nvidia-nvtx-cu12            12.1.105              pypi_0              pypi
opencv-python               4.12.0.88             pypi_0              pypi
opendatalab                 0.0.10                pypi_0              pypi
openmim                     0.3.9                 pypi_0              pypi
openssl                     3.6.1                 h35e630c_1          conda-forge
openxlab                    0.1.3                 pypi_0              pypi
ordered-set                 4.1.0                 pypi_0              pypi
oss2                        2.17.0                pypi_0              pypi
packaging                   24.2                  pypi_0              pypi
pandas                      2.3.3                 pypi_0              pypi
parso                       0.8.5                 pyhcf101f3_0        conda-forge
pexpect                     4.9.0                 pyhd8ed1ab_1        conda-forge
pickleshare                 0.7.5                 pyhd8ed1ab_1004     conda-forge
pillow                      11.3.0                pypi_0              pypi
pip                         25.2                  pyh8b19718_0        https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
platformdirs                4.5.0                 pypi_0              pypi
prompt-toolkit              3.0.52                pyha770c72_0        conda-forge
psutil                      7.2.1                 py310h139afa4_0     conda-forge
ptyprocess                  0.7.0                 pyhd8ed1ab_1        conda-forge
pure_eval                   0.2.3                 pyhd8ed1ab_1        conda-forge
pycocotools                 2.0.10                pypi_0              pypi
pycparser                   2.23                  pypi_0              pypi
pycryptodome                3.23.0                pypi_0              pypi
pygments                    2.19.2                pyhd8ed1ab_0        conda-forge
pyparsing                   3.2.5                 pypi_0              pypi
python                      3.10.19               hd994cfb_1_cpython  https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
python-dateutil             2.9.0.post0           pyhe01879c_2        conda-forge
python_abi                  3.10                  8_cp310             conda-forge
pytz                        2023.4                pypi_0              pypi
pyyaml                      6.0.3                 pypi_0              pypi
pyzmq                       27.1.0                py310h4f33d48_0     conda-forge
readline                    8.2                   h8c095d6_2          https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
requests                    2.28.2                pypi_0              pypi
rich                        13.4.2                pypi_0              pypi
scipy                       1.15.3                pypi_0              pypi
setuptools                  60.2.0                pypi_0              pypi
six                         1.17.0                pyhe01879c_1        conda-forge
stack_data                  0.6.3                 pyhd8ed1ab_1        conda-forge
sympy                       1.14.0                pypi_0              pypi
tabulate                    0.9.0                 pypi_0              pypi
termcolor                   3.1.0                 pypi_0              pypi
terminaltables              3.1.10                pypi_0              pypi
timm                        0.6.12                pypi_0              pypi
tk                          8.6.13                noxft_hd72426e_102  https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
tomli                       2.3.0                 pypi_0              pypi
torch                       2.3.1+cu121           pypi_0              pypi
torchvision                 0.18.1+cu121          pypi_0              pypi
tornado                     6.5.3                 py310h7c4b9e2_0     conda-forge
tqdm                        4.65.2                pypi_0              pypi
traitlets                   5.14.3                pyhd8ed1ab_1        conda-forge
triton                      2.3.1                 pypi_0              pypi
typing_extensions           4.15.0                pyhcf101f3_0        conda-forge
tzdata                      2025.2                pypi_0              pypi
urllib3                     1.26.20               pypi_0              pypi
wcwidth                     0.2.14                pyhd8ed1ab_0        conda-forge
wheel                       0.45.1                pyhd8ed1ab_1        https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
yapf                        0.43.0                pypi_0              pypi
zeromq                      4.3.5                 h387f397_9          conda-forge

**Comment by WuBai2005** [comment]
我找到了，我之前给mmcv打了个补丁！
先找self._use_replicated_tensor_module else self.module，再把这个改成getattr(self, '_use_replicated_tensor_module', False) else self.module

**Comment by WuBai2005** [comment]
路径是/home/pxh/miniconda3/envs/overlock/lib/python3.10/site-packages/mmcv/parallel/distributed.py

**Comment by WuBai2005** [comment]
欸，好像readme里面有提到
